server/core/src/main/java/dev/slimevr/util/LRUCache.kt

```diff
@@ -0,0 +1,112 @@
+package dev.slimevr.util
+
+import java.util.*
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * A thread-safe LRU (Least Recently Used) cache implementation using a LinkedHashMap
+ * for storing key-value pairs and a ReentrantLock for synchronization.
+ *
+ * @param capacity The maximum number of key-value pairs the cache can hold.
+ * @param <K> The type of keys.
+ * @param <V> The type of values.
+ */
+class LRUCache<K, V>(private val capacity: Int) {
+    private val cache: LinkedHashMap<K, V> = object : LinkedHashMap<K, V>(capacity, 0.75f, true) {
+        override fun removeEldestEntry(eldest: Map.Entry<K, V>?): Boolean {
+            return size > capacity
+        }
+    }
+    private val lock = ReentrantLock()
+
+    /**
+     * Get the value associated with the specified key.
+     *
+     * @param key The key to retrieve the value for.
+     * @return The value associated with the key, or null if the key is not in the cache.
+     */
+    fun get(key: K): V? {
+        lock.lock()
+        try {
+            return cache[key]
+        } finally {
+            lock.unlock()
+        }
```

Comment:
You should be able to replace all these try finallys with Kotlin's `withLock` extension method. This just automatically fills the try finally pattern.
```suggestion
        lock.withLock {
            return cache[key]
        }
```